lib/core/waitForCoreChainLockSyncFactory.js

```diff
@@ -0,0 +1,100 @@
+const { ChainLock } = require('@dashevo/dashcore-lib');
+
+const ZMQClient = require('@dashevo/dashd-zmq');
+
+const ensureBlock = require('./ensureBlock');
+
+/**
+ * Check that core is synced (factory)
+ *
+ * @param {ZMQClient} coreZMQClient
+ * @param {RpcClient} coreRpcClient
+ * @param {LatestCoreChainLock} latestCoreChainLock
+ * @param {BaseLogger} logger
+ *
+ * @returns {waitForCoreSync}
+ */
+function waitForCoreChainLockSyncFactory(
+  coreZMQClient,
+  coreRpcClient,
+  latestCoreChainLock,
+  logger,
+) {
+  /**
+   * Check that chainLock is synced
+   *
+   * @typedef waitForCoreChainLockSync
+   *
+   * @returns {Promise<LatestCoreChainLock>}
+   */
+  async function waitForCoreChainLockSync() {
+    await coreZMQClient.connect();
+
+    // By default will try to reconnect so we just log when this happen
```

Comment:
Can you send me a link to the code that reconnects on the `disconnect` event?

Reply:
The best way is to test it. Subscribe to connect:delay, connect, and reconnect. 
Else what look up zmq.Observer.Events for the events or zmq.Subscriber for the reconnect. 
The JS doc in ZMQ-dashd (or it's docsify) also documents it.

Reply:
- [SocketMonitor that allow us to get .on() on those events](https://github.com/dashevo/js-dashd-zmq/blob/master/lib/ZMQClient/methods/connect.js#L26)
- [ReconnectInterval as part of passed options from connect() method](https://github.com/zeromq/zeromq.js/blob/master/src/index.ts#L524)
- [As its binding on libzmq, the socks_connecter file display the reconnect implementation on connect:error](https://github.com/zeromq/libzmq/blob/master/src/socks_connecter.cpp#L273)

Reply:
So disconnection is default, but settings interval to -1 will overwrite default reconnection attempts.

Reply:
how many times we try to reconnect?

Reply:
We can have a backoff mechanism with an intervalMax : (see : [ZMQ_RECONNECT_IVL_MAX](http://api.zeromq.org/4-2:zmq-setsockopt#toc38) and [reconnectMaxInterval](https://zeromq.github.io/zeromq.js/classes/subscriber.html#reconnectmaxinterval))
But no limit by default, but implementation of a reconnect stop is actually pretty trivial as each connection tries is emited and listenable, a simple counter that would call .close() after X retries is something I can implement. 
I just need to know which rules we go for, what is our max, etc...